analytics/loggable_objects.go

```diff
@@ -0,0 +1,96 @@
+package analytics
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"net/http"
+)
+
+type RequestType string
+
+const (
+	COOKIE_SYNC RequestType = "cookie_sync"
+	AUCTION     RequestType = "auction"
+	SETUID      RequestType = "set_uid"
+	AMP         RequestType = "amp"
+)
+
+//Loggable object of a transaction at /openrtb2/auction endpoint
+type AuctionObject struct {
+	Type      RequestType
+	Status    int
+	Error     []error
+	Request   openrtb.BidRequest
+	Response  openrtb.BidResponse
+	UserAgent string
+}
+
+type AmpObject struct {
+	Type      RequestType
+	Status    int
+	Error     []error
+	Request   string
+	Response  string
+	UserAgent string
+	Origin    string
+}
+
+//Bid requests from each adapter. This information is available in /openrtb2/auction response but necessary to log in case there are errors
+type LoggableAdapterRequest struct {
+	Name     string
+	Requests string
+	Uri      string
+	Header   http.Header
+	Method   string
+}
+
+//Loggable object of a transaction at /setuid
+type SetUIDObject struct {
+	Type    RequestType
+	Status  int
+	Error   []error
+	Cookie  string
```

Comment:
Can this be structured?

It seems to me like (at minimum) you'd need:

- The Bidder being synced
- Did a sync exist for this bidder?

Other data may be relevant:

- How much time was left until this sync expired?
- Other syncs & expirations which exist

All that should be parsed out by the app code, though--not the analytics impls.